apps/passport-server/test/frogcrypto.spec.ts

```diff
@@ -0,0 +1,142 @@
+    }
+
+    expect(result).to.not.be.undefined;
+
+    identity = result.identity;
+  });
+
+  it("should be able to discover only public feeds", async function () {
+    const response = await requestListFeeds(
+      `${application.expressContext.localEndpoint}/frogcrypto/feeds`
+    );
+    expect(response.success).to.be.true;
+    const feeds = response.value?.feeds;
+    expect(feeds).to.not.be.undefined;
+    expect(feeds?.length).to.eq(1);
+    const feed = feeds?.[0] as FrogCryptoFeed;
+    expect(feed?.active).to.be.true;
+    expect(feed?.private).to.be.false;
+  });
+
+  it("should be able to get frog", async () => {
+    const feed = FROGCRYPTO_FEEDS[0];
+    expect(feed.active).to.be.true;
+    expect(feed.private).to.be.false;
+
+    const response = await getFrog(feed);
+    expect(response.success).to.be.true;
+
+    expect(response.value?.actions.length).to.eq(1);
+    const populateAction = response.value?.actions[0] as AppendToFolderAction;
+    expect(populateAction.type).to.eq(PCDActionType.AppendToFolder);
+    expect(populateAction.folder).to.eq("FrogCrypto");
+    expect(populateAction.pcds[0].type).to.eq(EdDSAFrogPCDPackage.name);
+
+    frogPCD = await EdDSAFrogPCDPackage.deserialize(populateAction.pcds[0].pcd);
+    expect(frogPCD.claim.data.ownerSemaphoreId).to.eq(
+      identity.commitment.toString()
+    );
+  });
+
+  it("should be able to get frog even if feed is private", async () => {
+    const feed = FROGCRYPTO_FEEDS[1];
+    expect(feed.active).to.be.true;
+    expect(feed.private).to.be.true;
+
+    const response = await getFrog(feed);
+    expect(response.success).to.be.true;
+
+    expect(response.value?.actions.length).to.eq(1);
+    const populateAction = response.value?.actions[0] as AppendToFolderAction;
+    expect(populateAction.type).to.eq(PCDActionType.AppendToFolder);
+    expect(populateAction.folder).to.eq("FrogCrypto");
+    expect(populateAction.pcds[0].type).to.eq(EdDSAFrogPCDPackage.name);
+
+    frogPCD = await EdDSAFrogPCDPackage.deserialize(populateAction.pcds[0].pcd);
+    expect(frogPCD.claim.data.ownerSemaphoreId).to.eq(
+      identity.commitment.toString()
+    );
+  });
+
+  it("should not be able to get frog if feed is inactive", async () => {
+    const feed = FROGCRYPTO_FEEDS[2];
+    expect(feed.active).to.be.false;
+    expect(feed.private).to.be.true;
+
+    const response = await getFrog(feed);
+    expect(response.success).to.be.false;
+    expect(response.error).to.include("not active");
+  });
+
+  async function getFrog(feed: FrogCryptoFeedConfig): Promise<PollFeedResult> {
+    MockDate.set(new Date());
+    const payload = JSON.stringify(createFeedCredentialPayload());
+    const response = await pollFeed(
+      application.expressContext.localEndpoint,
+      identity,
+      payload,
+      feed.id,
+      `${application.expressContext.localEndpoint}/frogcrypto/feeds`
+    );
+    MockDate.reset();
+
+    return response;
+  }
+});
```

Comment:
another test case to add here is one for issuance timeouts - i.e. that you can't request 2 frogs within some short amount of time.

Reply:
skipping coz it's superseded by the following PR